### ISSUE TITLE
Supporting BatchNorm

### ISSUE THREAD

**Issue by egretholier** [lucidrains/x-transformers#320]
Phil - How do you feel about adding an option to support BatchNorm in AttentionLayers? I am happy to contribute the code and test it on time series problems.

```
class BatchNorm(nn.Module):
    def __init__(self, dim):
        super().__init__()
        self.batchnorm = nn.BatchNorm1d(dim)

    def forward(self, x):
        return self.batchnorm(x.transpose(1, 2)).transpose(1, 2)

...

assert at_most_one_of(use_scalenorm, use_rmsnorm, use_dynamic_tanh, use_simple_rmsnorm, use_adaptive_layernorm, use_adaptive_rmsnorm, use_batchnorm), 'you can only use either scalenorm, rmsnorm, adaptive layernorm, adaptive rmsnorm, batchnorm, or simple rmsnorm'

...

elif use_batchnorm:
   norm_class = BatchNorm
```

Cons: 
1) Yet another argument to AttentionLayers
2) This is different from FFNBN in the [Leveraging Batch Normalization for Vision Transformers](https://openaccess.thecvf.com/content/ICCV2021W/NeurArch/papers/Yao_Leveraging_Batch_Normalization_for_Vision_Transformers_ICCVW_2021_paper.pdf) paper.

Pros:
1) Time series transformer papers are empirically claiming better performance. [A Transformer-based Framework for Multivariate Time Series Representation Learning](https://arxiv.org/abs/2010.02803) made the original claim and [A Time Series is Worth 64 Words: Long-term Forecasting with Transformers](https://arxiv.org/abs/2211.14730) made it popular.

**Comment by egretholier** [comment]
Update:
Here's the PR that implements use_batchnorm: https://github.com/lucidrains/x-transformers/pull/321

I verified that this works visually and empirically. For some strange reason, BatchNorm is working better for time series problems.

Excerpt from Zerveas and others 2021
> However, here we instead use batch normalization, because it can mitigate the effect of outlier values in time series, an issue that does not arise in NLP word embeddings.


<img width="886" alt="Image" src="https://github.com/user-attachments/assets/7286b864-6828-4a2a-aa8f-0ef5bd66b8fc" />
